databricks/koalas/indexes.py

```diff
@@ -961,6 +961,56 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def append(self, other):
+        """
+        Append a collection of Index options together.
+
+        Parameters
+        ----------
+        other : Index
+
+        Returns
+        -------
+        appended : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 5, 0, 5, 10, 5, 0, 10])
+        >>> kidx
+        Int64Index([10, 5, 0, 5, 10, 5, 0, 10], dtype='int64')
+
+        >>> kidx.append(kidx)
+        Int64Index([10, 5, 0, 5, 10, 5, 0, 10, 10, 5, 0, 5, 10, 5, 0, 10], dtype='int64')
+
+        Support for MiltiIndex
+
+        >>> kidx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y')])
+        >>> kidx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y')],
+                   )
+
+        >>> kidx.append(kidx)  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('a', 'x'),
+                    ('b', 'y')],
+                   )
+        """
+        if type(self) is not type(other):
+            raise NotImplementedError(
+                "append() between Index & MultiIndex currently is not supported")
+
+        sdf_self = self._internal.sdf
+        sdf_other = other._internal.sdf
+        sdf_appended = sdf_self.union(sdf_other)
```

Comment:
This doesn't follow the actual index:

```py
>>> kdf1
   a  b
a  1  4
b  2  5
c  3  6
>>> kdf2
   a   d
x  7  10
y  8  11
z  9  12

>>> kdf1.set_index('a').index.append(kdf2.set_index('d').index)
Int64Index([1, 2, 3, 7, 8, 9], dtype='int64')
```

whereas:

```py
>>> pdf1.set_index('a').index.append(pdf2.set_index('d').index)
Int64Index([1, 2, 3, 10, 11, 12], dtype='int64')
```

Reply:
@ueshin thanks for checking this out, i just fixed & added related test cases :)